hazelcast/src/main/java/com/hazelcast/core/MapLoader.java

```diff
@@ -26,12 +26,29 @@
  * it can be backed by any type of data store such as RDBMS, OODBMS, or simply
  * a file based data store.
  * <p/>
- * IMap.get(key) normally returns the value that is available in-memory. If the entry
- * doesn't exist in-memory, Hazelcast returns null. If a Loader implementation
- * is provided then, instead of returning null, Hazelcast will load the unknown entry by
- * calling the implementation's load (key) or loadAll(keys) methods. Loaded entries
- * will be placed into the distributed map and they will stay in-memory until they are
- * explicitly removed or implicitly evicted (if eviction is configured).
+ * {@link com.hazelcast.core.IMap#get(Object)} normally returns the value that is available in-memory. If the entry
+ * doesn't exist in-memory, Hazelcast returns null. If a Loader implementation is provided then, instead of returning
+ * null, Hazelcast will attempt to load the unknown entry by calling the implementation's {@link #load(Object)} or
+ * {@link #loadAll(Collection)}  methods. Loaded entries will be placed into the distributed map and they will stay
+ * in-memory until they are explicitly removed or implicitly evicted (if eviction is configured).
+ * <p/>
+ * MapLoader implementations are executed by a partition thread, therefore care should
+ * be taken not to block the thread with an expensive operation or an operation that may potentially
+ * never return, the partition thread does not time out the operation.  Whilst the partition thread is
+ * executing the MapLoader it is unable to respond to requests for data on any other structure that
+ * may reside in the same partition. For example a very slow MapLoader for one map could block a request
```

Comment:
may reside in the same partition, or in partitions mapped to the same partition thread.